Regression: line_length does not trigger error when ignores_function_declarations is enabled

### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Configuration:

```yml
line_length:
  warning: 100
  error: 100 
  ignores_function_declarations: true
```

Source file:
```swift
func test() {
    try FileManager.default.url(for: .cachesDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
}
```

Xcode version:
```bash
% xcodebuild -version
Xcode 26.1.1
Build version 17B100
```

I tested portable versions from GitHub. When using `0.62.2`:
```bash
./swiftlint --config ./.swiftlint.yaml
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

Latest working version was `0.59.1`:
```bash
./swiftlint --config ./.swiftlint.yaml
<path removed>/test.swift:2:1: error: Line Length Violation: Line should be 100 characters or less; currently it has 110 characters (line_length)
Done linting! Found 1 violation, 1 serious in 1 file.
```

Setting `ignores_function_declarations` to false resolves the issue on `0.62.2`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -34,6 +34,12 @@
 
 ### Bug Fixes
 
+* Fix `line_length` rule incorrectly ignoring function bodies when
+  `ignores_function_declarations` is enabled. The option should only ignore
+  function declarations, not their implementation.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#6347](https://github.com/realm/SwiftLint/issues/6347)
+
 * Fix false positives in `vertical_whitespace_between_cases` rule when cases are
   interleaved with compiler directives like `#if`/`#else`/`#endif`.  
   [SimplyDanny](https://github.com/SimplyDanny)

--- a/Source/SwiftLintBuiltInRules/Rules/Metrics/LineLengthRule.swift
+++ b/Source/SwiftLintBuiltInRules/Rules/Metrics/LineLengthRule.swift
@@ -147,7 +147,7 @@ private extension LineLengthRule {
 
 // MARK: - Helper Visitors for Pre-computation
 
-// Visitor to find lines spanned by function declarations
+// Visitor to find lines spanned by function declaration signatures
 private final class FunctionLineVisitor: SyntaxVisitor {
     let locationConverter: SourceLocationConverter
     var lines = Set<Int>()
@@ -158,20 +158,32 @@ private final class FunctionLineVisitor: SyntaxVisitor {
     }
 
     override func visitPost(_ node: FunctionDeclSyntax) {
-        collectLines(from: node)
+        collectLines(
+            from: node.positionAfterSkippingLeadingTrivia,
+            to: node.genericWhereClause?.endPositionBeforeTrailingTrivia
+                ?? node.signature.endPositionBeforeTrailingTrivia
+        )
     }
 
     override func visitPost(_ node: InitializerDeclSyntax) {
-        collectLines(from: node)
+        collectLines(
+            from: node.positionAfterSkippingLeadingTrivia,
+            to: node.genericWhereClause?.endPositionBeforeTrailingTrivia
+                ?? node.signature.endPositionBeforeTrailingTrivia
+        )
     }
 
     override func visitPost(_ node: SubscriptDeclSyntax) {
-        collectLines(from: node)
+        collectLines(
+            from: node.positionAfterSkippingLeadingTrivia,
+            to: node.genericWhereClause?.endPositionBeforeTrailingTrivia
+                ?? node.returnClause.endPositionBeforeTrailingTrivia
+        )
     }
 
-    private func collectLines(from node: any SyntaxProtocol) {
-        let startLocation = locationConverter.location(for: node.positionAfterSkippingLeadingTrivia)
-        let endLocation = locationConverter.location(for: node.endPositionBeforeTrailingTrivia)
+    private func collectLines(from startPosition: AbsolutePosition, to endPosition: AbsolutePosition) {
+        let startLocation = locationConverter.location(for: startPosition)
+        let endLocation = locationConverter.location(for: endPosition)
         for line in startLocation.line...endLocation.line {
             lines.insert(line)
         }

--- a/Tests/BuiltInRulesTests/LineLengthRuleTests.swift
+++ b/Tests/BuiltInRulesTests/LineLengthRuleTests.swift
@@ -45,6 +45,20 @@ final class LineLengthRuleTests: SwiftLintTestCase {
             }
             """),
     ]
+    private let longFunctionCalls = [
+        Example("""
+            superDuperLongFunctionCall(a: "A", b: "B", c: "C", d: "D", e: "E", f: "F", \
+            g: "G", h: "H", i: "I", j: "J", k: "K", l: "L", m: "M", n: "N", o: "O", p: "P", \
+            q: "Q", r: "R", s: "S", t: "T", u: "U", v: "V", w: "W", x: "X", y: "Y", z: "Z")
+            """),
+        Example("""
+            func test() {
+                let _ = superDuperLongFunctionCall(a: "A", b: "B", c: "C", d: "D", e: "E", f: "F", \
+                g: "G", h: "H", i: "I", j: "J", k: "K", l: "L", m: "M", n: "N", o: "O", p: "P", \
+                q: "Q", r: "R", s: "S", t: "T", u: "U", v: "V", w: "W", x: "X", y: "Y", z: "Z")
+            }
+            """),
+    ]
 
     private let longComment = Example(String(repeating: "/", count: 121) + "\n")
     private let longBlockComment = Example("/*" + String(repeating: " ", count: 121) + "*/\n")
@@ -118,8 +132,10 @@ final class LineLengthRuleTests: SwiftLintTestCase {
 
     func testLineLengthWithIgnoreFunctionDeclarationsEnabled() {
         let baseDescription = LineLengthRule.description
-        let nonTriggeringExamples = baseDescription.nonTriggeringExamples + longFunctionDeclarations
-        let description = baseDescription.with(nonTriggeringExamples: nonTriggeringExamples)
+        let description = baseDescription.with(
+            nonTriggeringExamples: baseDescription.nonTriggeringExamples + longFunctionDeclarations,
+            triggeringExamples: longFunctionCalls
+        )
 
         verifyRule(
             description,

